src/invidious/routes/before_all.cr

```diff
@@ -100,6 +88,24 @@ module Invidious::Routes::BeforeAll
       end
     end
 
+    unregistered_path_whitelist = {"/", "/login", "/licenses", "/privacy"}
+    if CONFIG.login_only && !env.get?("user") && !unregistered_path_whitelist.includes?(env.request.path)
+      env.response.headers["Location"] = "/login"
+      haltf env, status_code: 302
+    end
```

Comment:
```suggestion
    unregistered_path_whitelist = {
      "/api/v1/comments/jNQXAC9IVRw",
      "/api/v1/stats",
      "/api/v1/videos/dQw4w9WgXcQ",
      "/feed/webhook/v1:",
      "/licenses",
      "/login",
      "/opensearch.xml",
      "/privacy"
      }

    if CONFIG.login_only && !env.get?("user") && !unregistered_path_whitelist.any? { |r| env.request.path.starts_with? r }
      env.response.headers["Location"] = "/login"
      haltf env, status_code: 401
    end
```

- Changing it to this comes with the advantage, that the login page is opened first.
- The API whitelisting allows third-clients to connect and add the instance and login to an existing account
- Changed the status code from 302 (Found) to 401 (Unauthorized) which is more appropriate in this case.